yaml/shared/src/main/scala/org/virtuslab/yaml/internal/load/reader/Tokenizer.scala

```diff
@@ -109,9 +108,22 @@ private[yaml] class Scanner(str: String) extends Tokenizer {
     val name = readAnchorName()
     (name, pos)
 
-  private def parseAnchor() =
-    val (name, pos) = parseAnchorName()
-    Token(Anchor(name), pos)
+  private def parseAnchor(): List[Token] =
+    val (name, anchorPos) = parseAnchorName()
+    val nexTokens         = getNextTokens()
+
+    val anchorToken = Token(Anchor(name), anchorPos)
+    nexTokens match {
+      case (Token(_: MappingStart.type, _) |
+          Token(_: FlowMappingStart.type, _)) :: Token(_: MappingKey.type, _) :: rest =>
+        ctx.removeLastIndent()
+        ctx.addIndent(anchorPos.column)
```

Comment:
So here we have to adjust the indentation level because that added by the mapping isn't the right one, correct?

Reply:
Yes, you are right. The anchor position indicates the mapping indent.